pkg/asset/installconfig/azure/session.go

```diff
@@ -224,11 +226,14 @@ func newSessionFromCredentials(cloudEnv azureenv.Environment, credentials *Crede
 		AADEndpoint:  cloudEnv.ActiveDirectoryEndpoint,
 	}
 	c.Resource = cloudEnv.TokenAudience
-	authorizer, err := c.Authorizer()
+
+	cred, err := azidentity.NewClientSecretCredential(credentials.TenantID, credentials.ClientID, credentials.ClientSecret, nil)
 	if err != nil {
-		return nil, errors.Wrap(err, "failed to get client credentials authorizer")
+		return nil, errors.Wrap(err, "failed to get client credentials secret")
 	}
 
+	authorizer := azidext.NewTokenCredentialAdapter(cred, []string{azidext.DefaultManagementScope})
```

Comment:
I think we will also want a comment here about how/why we are using `azidext` because we will eventually plan on removing it once all the clients are from the `/sdk` modules.